### PR TITLE
Fix temporary sync errors if there's weird settings stored in account data

### PR DESCRIPTION
**Pull request by turt2live**
When https://github.com/matrix-org/matrix-react-sdk/pull/8798 made its changes to the `SettingsStore` it subtly changed how `getLevelOrder` actually worked. The previous behaviour was a simple `if (ordered) { return orderedLevels; } else { return DEFAULTS; }` but the new behaviour added a check for if there was only ever one supported level, in which case it'd return that single level instead of the defaults. This is the correct behaviour and should in fact be enforced as it is, however it had a weird knock-on effect.

Earlier, I noticed my sync loop was a bit slow. This was found in the console:
```
rageshake.ts:73 Caught /sync error Error: Level "account" for setting "useCompactLayout" is not prioritized
    at SettingsStore_SettingsStore.getValueAt (SettingsStore.ts:363:36)
    at localizedCallback (SettingsStore.ts:178:51)
    at WatchManager_WatchManager.notifyUpdate (WatchManager.ts:72:13)
    at MatrixClient.<anonymous> (AccountSettingsHandler.ts:69:31)
    at MatrixClient.emit (events.js:158:1)
    at MatrixClient.emit (typed-event-emitter.ts:61:22)
    at sync.ts:1170:28
    at Array.forEach (<anonymous>)
    at SyncApi.processSyncResponse (sync.ts:1159:20)
    at SyncApi.doSync (sync.ts:920:24)
```

The error only shows up if `getLevelOrder` returns levels which aren't supported by the level provided to `getLevelAt` (called implicitly through a `WatchManager` update from the `AccountSettingsHandler` when it detects a change in account data). For the particular setting involved, the `useCompactLayout` setting has *never* supported the `account` level so it's surprising to see it come up in that context - I probably have it lingering in my account data from a previous test or something.

So, when the account data comes through and triggers an internal notification to the watch manager, it tries to read the value of the setting from a wrapped callback in order to provide more information to the downstream watching code. This read fails because of the weird setting name in an unexpected location. Previously, it wouldn't have been an issue because it was assumed that the default levels would cover the setting's off-spec behaviour, but with the change to returning the single level in the case of single-level-supported settings it ended up accidentally breaking settings like this which only listen at the `device` level. 

----

Ultimately, I feel this is a super rare case that doesn't need a regression test. We should have more tests on the settings store generally, though I also would like to `Ctrl A + Delete` it from the project (replacing it with a new one with proper types, a la https://github.com/matrix-org/matrix-react-sdk/pull/5669 )

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix temporary sync errors if there's weird settings stored in account data ([\#8857](https://github.com/matrix-org/matrix-react-sdk/pull/8857)).

### CODE REVIEW

File: src/settings/SettingsStore.ts [hunk 1]
```diff
@@ -516,6 +524,23 @@ export default class SettingsStore {
         return LEVEL_HANDLERS[level].isSupported();
     }
 
+    /**
+     * Determines if a setting supports a particular level.
+     * @param settingName The setting name.
+     * @param level The level.
+     * @returns True if supported, false otherwise. Note that this will not check to see if
```
Comment (SimonBrandner) [inline src/settings/SettingsStore.ts:531]: I think the types should also be specified here in curly brackets?

**Review by SimonBrandner — APPROVED** [review]
Otherwise LGTM

File: src/settings/SettingsStore.ts (hunk 1 above)
Comment (turt2live) [reply src/settings/SettingsStore.ts:531]: That's only really applicable if we don't have TypeScript to tell us the types :)

Comment (turt2live) [reply src/settings/SettingsStore.ts:531]: (it's easier to not delete the types when converting something from JS->TS, which is why the other functions have the types still)

Comment (SimonBrandner) [reply src/settings/SettingsStore.ts:531]: Ah, ok, I've actually never really thought about this much - thank you for the explanation!

**Review by SimonBrandner — APPROVED** [review]

**Pull request merged**